src/Microsoft.AspNetCore.Authentication/AuthenticationMiddleware.cs

```diff
@@ -31,34 +32,47 @@ public AuthenticationMiddleware(RequestDelegate next, IAuthenticationSchemeProvi
 
         public async Task Invoke(HttpContext context)
         {
-            context.Features.Set<IAuthenticationFeature>(new AuthenticationFeature
+            var span = AuthenticationEventSource.Log.AuthenticationMiddlewareStart(context.TraceIdentifier, context.Request.Path);
```

Comment:
Any reason not to just pass the context and let the event extract the trace identifier/path as needed?

Reply:
That's reasonable too, it depends on the specific event. When I started this, I didn't have the pattern of `internal` with `[NonEvent]` and `private` with `[Event]` so I had to have only primitive types in the signature (required by the `[Event]` attribute)